content/webapp/components/EventsSearchResults/index.tsx

```diff
@@ -105,7 +105,7 @@ const EventsSearchResults: FunctionComponent<Props> = ({
               <EventInformation>
                 {/* Past events that are available online don't have a status indicator
                 as they display online availability information. */}
-                {isPast && !event.isAvailableOnline && (
+                {times.length > 0 && isPast && !event.isAvailableOnline && (
```

Comment:
What situations result in `times` being empty?

Reply:
I wrote it quickly in the description but Prismic doesn't require anything, every field is optional. So if a times block exists but isn't filled in and the event is published, times will come back as an empty array.

<img width="1220" alt="Screenshot 2024-01-04 at 15 14 47" src="https://github.com/wellcomecollection/wellcomecollection.org/assets/110461050/8099bcfc-c33f-4daa-9760-130ae4ca02ef">

Now if this should be handled before we get to this stage we can discuss, like maybe it should return undefined instead of an empty array?

Reply:
> Tweak a condition that was breaking the page when an event had no times set
Ah sorry, missed that! 

I wonder if the content API should rationalise this somehow instead of dealing with it here?

Reply:
So that's an interesting point that @agnesgaroux and I talked about when it all started; what gets done on the API vs what gets done on the FE. My post-holiday self can't seem to find traces of that chat though @agnesgaroux we had documented it somewhere I believe, but not in https://www.notion.so/Content-API-129383af9efa40e9815c3450aeebb389?

Reply:
I would favour pushing this as far back as possible, ideally being able to express which fields can be optional when in the editor. I imagine this isn't possible though?

Reply:
I'm not sure what you mean, so sorry if I'm repeating things you know but [Prismic doesn't allow for required fields](https://prismic.io/blog/required-fields) (big sigh imo), so we can't trust that it'll always be filled in even if we add something in the label to indicate it _should_ be filled. Is that what you meant?

Reply:
> [Prismic doesn't allow for required fields](https://prismic.io/blog/required-fields)

I didn't realise that was a for prismic in general, but perhaps just not enforced in this case. That is a bit surprising!

> I would favour pushing this as far back as possible

IMO the model of what should be optional or not is best dealt with close to the editor because as it gets pushed downstream you end up with a situation like this where you have to account for unexpected states.

Reply:
It definitely has been a thing, where pages that were normally running smoothly would suddenly broken pages because a niche field hadn't been filled in for the first time and we'd have to go and add a condition or change our transformers. It was necessary when we were only using the Prismic API, but now we could consider changing tack. I know there were arguments on both sides around this though, so maybe good to bring up with the team? Maybe not in the workshop that I think will be much higher level though... In Slack?